packages/yup/src/dateRange.js

```diff
@@ -0,0 +1,52 @@
+import moment from 'moment';
+
+export default function({ min, max, format = 'MM/DD/YYYY' }, msg) {
+  const minDate = moment(
+    min,
+    ['MM/DD/YYYY', format, 'MMDDYYYY', 'YYYYMMDD'],
+    true
+  );
+
+  const maxDate = moment(
+    max,
+    ['MM/DD/YYYY', format, 'MMDDYYYY', 'YYYYMMDD'],
+    true
+  );
+
+  // Can't use arrow function because we rely on 'this' referencing yup's internals
+  return this.test({
+    name: 'dateRange',
+    exclusive: true, // Validation errors don't stack
+    // NOTE: Intentional use of single quotes - yup will handle the string interpolation
+    message:
+      msg ||
+      `Date Range must be between ${minDate.format(
```

Comment:
Can this message change based on the specific check which failed? If the start and end dates are both between the min and max but the end comes before the start then the validation will indicate an error but the message will not be correct.

Reply:
We can just update it to make sure both dates are between the specified dates.

How the default `yup.date()` works is you can specify `yup.date().min().max()` and it will say what value failed but if they both failed it won't say anything about between.